src/components/geojson-map.vue

```diff
@@ -317,18 +333,77 @@ const moveOverFeature = (event) => {
   mapContainer.value.style.cursor = hit ? 'pointer' : '';
 };
 
+const forEachFeatureNearPixel = (pixel, radius, callback) => {
+  // Start with a boundary box.
+  const corners = [
+    [pixel[0] - radius, pixel[1] - radius], // top-left
+    [pixel[0] + radius, pixel[1] + radius] // bottom-right
+  ];
+  const extent = boundingExtent(corners.map(p =>
+    mapInstance.getCoordinateFromPixel(p)));
+
+  // For features within `extent`, filter for those that are actually within
+  // `radius` pixels of `pixel`.
+  const coordinate = mapInstance.getCoordinateFromPixel(pixel);
+  const r2 = radius ** 2;
+  featureSource.forEachFeatureIntersectingExtent(extent, (feature) => {
+    const closest = mapInstance.getPixelFromCoordinate(
+      feature.getGeometry().getClosestPoint(coordinate)
+    );
+    if ((closest[0] - pixel[0]) ** 2 + (closest[1] - pixel[1]) ** 2 <= r2)
+      callback(feature);
+  });
+};
+
+const overlapRadius = clusterSource.getDistance().value / 2;
+
+const getHits = (pixel) => {
+  const hits = mapInstance.getFeaturesAtPixel(pixel, hitDetectionOptions);
+  if (hits.length === 0) return hits;
+
+  if (clusterLayer.isVisible()) {
+    const cluster = hits.find(isCluster);
+    if (cluster != null) return [cluster];
+  }
+
+  // getFeaturesAtPixel() above usually works great, but when features overlap,
+  // it only seems to return the feature on top. Here, we try to detect
+  // additional, overlapping hits by searching for features near `pixel`. As we
+  // do so, we avoid duplicates.
+  const ids = hits.reduce((set, hit) => set.add(hit.getId()), new Set());
+  forEachFeatureNearPixel(pixel, overlapRadius, (feature) => {
+    if (!ids.has(feature.getId())) hits.push(feature);
+  });
+
+  return hits;
+};
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// SELECT FEATURE
+
+let selectedId;
+
+// Converts an OpenLayers Feature object to a simple object before emitting it.
+// That way, the parent component never has to interact with OpenLayers.
+const featureToObject = (feature) =>
+  ({ id: feature.getId(), properties: feature.getProperties() });
+
 // Selects an individual feature (not a cluster) or deselects the selected
 // feature (if feature is `null`).
-const selectFeature = (feature) => {
+const selectFeature = (feature, emitChange = true) => {
   const id = feature?.getId();
   if (id === selectedId) return;
 
-  if (featureLayer instanceof WebGLVectorLayer)
-    featureLayer.updateStyleVariables({ selectedId: id ?? '' });
+  mapInstance.getLayers().forEach(layer => {
```

Comment:
Maybe this would be better?

```suggestion
  for (const layer of [featureLayer, clusterLayer]) {
```